src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethodInvoker.cs

```diff
@@ -17,8 +14,23 @@ public DynamicMethodInvoker(DynamicMethod dynamicMethod)
 
         public unsafe object? InvokeUnsafe(object? obj, IntPtr* args, BindingFlags invokeAttr)
         {
-            // Todo: add strategy for calling IL Emit-based version
-            return _dynamicMethod.InvokeNonEmitUnsafe(obj, args, invokeAttr);
+            // Always use the slow path; the Emit-based fast path can be added but in general dynamic
```

Comment:
We should either remove `DynamicMethodInvoker` or add the fast path to it and test it appropriately. It's not clear to me when the "hybrid" use of a dynamic method + loosely-typed invoke is used. Normally a dynamic method is create for performance, so the corresponding delegate is invoked directly.

Reply:
I would expect it to be used in complex systems with a lot of dynamic code, like powershell.

Do we need a separate implementation of the Invoker for DynamicMethods? Can we share the Invoker implementation from regular RuntimeMethodInfo here? The Invoker should really only need a method signature to work.

Reply:
I'll hook up emit support for DynamicMethod then.

We could share the same invoker but due to code sharing with Mono I believe I'd need to add a virtual method to `MethodBase` in order to call the native\non-emit method from the invoker. Doable, but could impact perf a tiny bit.

Reply:
It is fine to use ifdefs to workaround the temporary sharing problems with Mono.

Reply:
OK using ifdefs will work. In that case I can also avoid passing the unused `argsForTemporaryMonoSupport` parameter around.

Reply:
For this to work, I'm moving the `InvokeNonEmitUnsafe()` that is on `MethodInfo`, `ConstructorInfo` (and potentially `DynamicMethod` which is the issue) to a CoreCLR-specific and Mono-specific helper on the invoker classes.